### ISSUE TITLE
[Feature Request] - Backup and Restore of watch status.

### ISSUE THREAD

**Issue by devanteweary** [luigi311/JellyPlex-Watched#130]
Hey Luigi! What's up man??

I forgot to make this an actual "issue" so here it is.

Basically a feature suggestion to backup and restore each user's watch state.

Maybe the option to select Plex, JF, or both.  

Alright see ya!

**Comment by luigi311** [comment]
Backup should be easy to handle, i can make it just save the dictionary of the watch list to a file if that file does not exist already that way the backup is only generated on the first run and not replaced after every sync. 

Restoring it though might be weird since it will probably need a ENV variable such as RESTORE set to true that will change the path this takes.

Restore can be handled two ways

1. By generating the current watch status and marking anything that is in the new dictionary and not in the backup as unwatched/partially watched and then exits as if the RUN_ONLY_ONCE variable is set. 

2. Wipe that entire users played list and then running the backup dictionary as if it is going through the normal flow and marking them as watched

1 is a lot less destructive in that it wont have to wipe everything and potentially leave you with nothing watched but that requires more functions being made that might have bugs.  
2 has the added benefit just requiring a function that wipes the users watch history as it will use the existing ones for handling the marking for the restore portion.

Either way i wont be implementing this just yet as i want to marinate on which implementation i want to take for the restoring process.

**Comment by gaming09** [comment]
that would be awesome

**Comment by marshalleq** [comment]
Yeah, the first thing I thought of when I started using this, is the one missing thing it has is just a backup locally to somewhere else type of feature.  The amount of times this is all I've needed over the past decade is crazy - and I'm back here again to check if it's in yet, because today this would have saved me.  Now I'm back to restoring from an old app to a new app to get the 'most' of my watch status.  Probably wouldn't be too hard to do either and a lot smaller than an actual database backup.
